src/core/catalog.js

```diff
@@ -1203,7 +1205,9 @@ class Catalog {
         }
         visitedNodes.put(currentNode);
 
-        const obj = await xref.fetchAsync(currentNode);
+        const obj =
+          (await this.pageDictCache.get(currentNode)) ||
+          xref.fetchAsync(currentNode);
```

Comment:
Sorry, but this doesn't look correct w.r.t. the placement of the parenthesis. (This should, hopefully, case test-failures as-is.)
```suggestion
        const obj =
          await (pageDictCache.get(currentNode) || xref.fetchAsync(currentNode));
```

Here I'm assuming that a local variable is added, similar to the other caches: https://github.com/mozilla/pdf.js/blob/e0fc1a341a9917e1cdd15f2d4b91cee3f448bea0/src/core/catalog.js#L1185-L1187

Reply:
Oops yeah. I didn't put parens in at all and eslint added them, apparently I don't know what the precedence of await is.

It actually does work and the tests pass because
- on the first pass it is looking at the /Pages dict, that isn't a Dict so the fact that obj is a Promise not the actual object doesn't matter, and obj isn't used again, and
- after that everything we want to load has been added to the cache

however you're right that this is clearly incorrect.

The local variable is a style thing but I have no problem with making that consistent as well